var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py

```diff
@@ -71,11 +72,12 @@ class PyPennylaneLightningKokkos(CMakePackage, PythonExtension, CudaPackage, ROC
     depends_on("py-pybind11", type="link")
     depends_on("py-pip", type="build")
     depends_on("py-wheel", type="build")
-    depends_on("py-pennylane@0.30:", type=("build", "run"), when="@0.31:")
-    depends_on("py-pennylane@0.28:", type=("build", "run"), when="@:0.30.0")
-    depends_on("py-pennylane-lightning@0.31:~kokkos", type=("build", "run"), when="@0.31.0:")
-    depends_on("py-pennylane-lightning@0.30:~kokkos", type=("build", "run"), when="@0.30.0:")
-    depends_on("py-pennylane-lightning@0.28:0.29~kokkos", type=("build", "run"), when="@0.29.1")
+    depends_on("py-pennylane@0.28:", type=("build", "run"), when="@:0.30")
+    depends_on("py-pennylane@0.31:", type=("build", "run"), when="@0.31:")
+    depends_on("py-pennylane-lightning~kokkos", type=("build", "run"), when="@:0.31")
+    depends_on("py-pennylane-lightning@0.28:0.29", type=("build", "run"), when="@0.29.1")
+    for v in range(30, 33):
+        depends_on(f"py-pennylane-lightning@0.{v}:", type=("build", "run"), when=f"@0.{v}.0:")
```

Comment:
I'm sure I've asked this before, but I don't see `pennylane-lightning` listed as a dependency in `setup.py`. Is it documented somewhere?

Reply:
It is a "tangential" dependency - if I may say so - as I tried to explain [here](https://github.com/spack/spack/pull/39688#discussion_r1310232623). That said, I think we can remove the version pinning statements, which are already in the `py-pennylane` file.